pkg/db/gateway_envelope.go

```diff
@@ -78,6 +78,45 @@ func InsertGatewayEnvelopeAndIncrementUnsettledUsage(
 	)
 }
 
+// InsertGatewayEnvelope inserts a gateway envelope into the database.
+// It does not increment the unsettled usage and congestion counters.
+// Used exclusively for reserved topics.
+//
+// This function runs inside a managed transaction created by RunInTxWithResult().
+//
+// Steps:
+//  1. Calls InsertGatewayEnvelopeWithChecksTransactional() to insert the envelope,
+//     automatically creating any missing partitions if needed.
+func InsertGatewayEnvelope(
```

Comment:
you do not need this function. There is no need for a transaction. You are only executing a single statement. You should use `InsertGatewayEnvelopeWithChecksStandalone`